File: JavaScript/eliuds-eggs.js
const toBinary = (number) => (number >>> 0).toString(2);

const eggCount = (displayValue) => {
  let binary = toBinary(displayValue);
  let count = binary.split("").filter(bit => bit === "1").length;

  return count;
};

const displayValues = [
  0,
  1,
  4,
  16,
  13,
  89,
  2_000_000_000,
];

displayValues.forEach(value => {
  console.log(`Value: ${value}`);
  console.log(`\tBinary: ${toBinary(value)}`);
  console.log(`\tEgg Count: ${eggCount(value)}`);
});

File: C#/Eliuds Eggs/EliudsEggs.cs
using System;
using System.Linq;

public static class EliudsEggs
{
    public static string ToBinary(int value)
    {
        return Convert.ToString(value, 2);
    }

    public static int EggCount(int encodedCount)
    {
        string binary = ToBinary(encodedCount);
        return binary.Where(bit => bit == '1').ToArray().Length;
    }
}